libs/ledger/include/ledger/dag/dag_hash.hpp

```diff
@@ -44,64 +29,33 @@ struct DAGHash
 {
   using ConstByteArray = byte_array::ConstByteArray;
 
-  DAGHash() = default;
-
-  explicit DAGHash(ConstByteArray h)
-    : hash{std::move(h)}
-  {}
-
-  explicit DAGHash(ConstByteArray h, char t)
-    : hash{std::move(h)}
-    , type{t}
-  {}
-
-  explicit operator ConstByteArray()
+  enum class Type : int8_t
   {
-    return hash;
-  }
-
-  bool empty() const
-  {
-    return hash.empty();
-  }
-
-  bool operator==(DAGHash const &other) const
-  {
-    return hash == other.hash;
-  }
+    NODE = 0,
+    EPOCH
+  };
 
-  bool operator>(DAGHash const &rhs) const
-  {
-    return hash > rhs.hash;
-  }
+  DAGHash() = default;
+  explicit DAGHash(ConstByteArray h);
+  explicit DAGHash(ConstByteArray h, Type t);
```

Comment:
Sorry, missed this the first time - explicit is only meaningful on constructors which take a single argument.

Reply:
fixed, thank you :)